vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java

```diff
@@ -1073,6 +1073,10 @@ public static VMwareDVSPvlanConfigSpec createDVPortPvlanConfigSpec(int vlanId, i
     }
 
     public static VmwareDistributedVirtualSwitchVlanSpec createDVPortVlanSpec(Integer vlanId, String vlanRange) {
+        if (vlanId != null && vlanId == 4095){
```

Comment:
@Spaceman1984 Have you tested this?

this method `createDVPortVlanSpec()` is called only when `vlanId` is null, so if check here is always 'false' and the underlying code never executes. Please check and try to fix this in the appropriate top level method in the call stack.

Reply:
This method is not olnly called when vlanid is null.
If vid is not null, spvlanid would be evaluated and if spvlanid is null, the method will be called.
```
if (vid == null || spvlanid == null) {
    vlanspec = createDVPortVlanSpec(vid, vlanRange);
    ...
}
```
I tested this as in the description.